src/main/java/uk/co/real_logic/benchmarks/rtt/Configuration.java

```diff
@@ -0,0 +1,543 @@
+     * @throws NullPointerException if {@code properties == null}
+     */
+    public static Configuration fromProperties(final Map<String, String> properties)
+    {
+        requireNonNull(properties);
+
+        final Builder builder = new Builder();
+        if (isPropertyProvided(properties, WARM_UP_ITERATIONS_PROP_NAME))
+        {
+            builder.warmUpIterations(parseInt(properties, WARM_UP_ITERATIONS_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, WARM_UP_MESSAGES_PROP_NAME))
+        {
+            builder.warmUpNumberOfMessages(parseInt(properties, WARM_UP_MESSAGES_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, WARM_UP_MESSAGES_PROP_NAME))
+        {
+            builder.warmUpNumberOfMessages(parseInt(properties, WARM_UP_MESSAGES_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, ITERATIONS_PROP_NAME))
+        {
+            builder.iterations(parseInt(properties, ITERATIONS_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, BURST_SIZE_PROP_NAME))
+        {
+            builder.burstSize(parseInt(properties, BURST_SIZE_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, MESSAGE_SIZE_PROP_NAME))
+        {
+            builder.messageSize(parseInt(properties, MESSAGE_SIZE_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, SENDER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.senderIdleStrategy(parseIdleStrategy(properties, SENDER_IDLE_STRATEGY_PROP_NAME));
+        }
+        if (isPropertyProvided(properties, RECEIVER_IDLE_STRATEGY_PROP_NAME))
+        {
+            builder.receiverIdleStrategy(parseIdleStrategy(properties, RECEIVER_IDLE_STRATEGY_PROP_NAME));
+        }
+        builder.numberOfMessages(parseInt(properties, MESSAGES_PROP_NAME));
+        builder.messageProviderClass(parseClass(properties, MESSAGE_PROVIDER_PROP_NAME, MessageProvider.class));
+        return builder.build();
+    }
+
+    private static int checkMinValue(final int value, final int minValue, final String prefix)
+    {
+        if (value < minValue)
+        {
+            throw new IllegalArgumentException(prefix + " cannot be less than " + minValue + ", got: " + value);
+        }
+        return value;
+    }
+
+    private static Class<? extends MessageProvider> validateMessageProviderClass(
+        final Class<? extends MessageProvider> klass)
+    {
+        requireNonNull(klass, "MessageProvider class cannot be null");
+        if (klass.isInterface() || Modifier.isAbstract(klass.getModifiers()))
+        {
+            throw new IllegalArgumentException("MessageProvider class must be a concrete class");
+        }
+        try
+        {
+            final Constructor<? extends MessageProvider> constructor = klass.getConstructor();
+            if (Modifier.isPublic(constructor.getModifiers()))
+            {
+                return klass;
+            }
+        }
+        catch (NoSuchMethodException e)
```

Comment:
No use of final. We should add checkstyle to the benchmarks.

Reply:
Will do.

Reply:
Added Checkstyle.